### PR TITLE
fix(#8096): block admins from updating own password

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

#8096 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8096-block-admins-from-updating-password/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8096-block-admins-from-updating-password/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8096-block-admins-from-updating-password/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by m5r** [comment]
I don't know whether it's the desired behavior but I managed to change the admin password from the user table in the admin app
I noticed we have API code specific to this scenario in https://github.com/medic/cht-core/blob/master/shared-libs/settings/src/index.js#L155-L167, if we're deprecating this use case we could block it server-side and remove some of this code

**Comment by garethbowen** [comment]
Excellent point, thanks @m5r !

I've updated the API with the same premise of blocking updating admin passwords which means throwing a 400 error when this is attempted through the API. This means users can't brick their install.

One day if we figure out how to update the env var in the container then we can allow this again, but for now this is safer...

**Review by m5r — APPROVED** [review]

File: tests/e2e/upgrade/admin-user.wdio-spec.js [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-
-    await utils.request({
-      path: `/api/v1/users/${adminUser.username}`,
-      method: 'POST',
-      body: { password: newPassword }
-    });
-
-    await loginPage.login({ username: adminUser.username, password: newPassword, adminApp: true });
-  });
-});
```
Comment (garethbowen) [inline tests/e2e/upgrade/admin-user.wdio-spec.js:34]: @dianabarsan This test is literally trying to change the admin password which is no longer possible as of this PR. The functionality being testing is a duplicate of an integration test we have so I'm not clear on why we test it in the upgrade suite too. Do you remember? Was it actually testing the password change or was it also doing something else?

Comment (dianabarsan) [reply tests/e2e/upgrade/admin-user.wdio-spec.js:34]: Thanks @garethbowen . I think it served as a check that new admins can still get created and login. I think keeping the test that just creates the new admin user and allows them to login to the admin app should be ok?

Comment (garethbowen) [reply tests/e2e/upgrade/admin-user.wdio-spec.js:34]: Thanks! Added it back and it passed. Too easy.

**Pull request merged**